# wage equation

### модель 1 

спецификация 
$$ 
log(wage) = \beta_0 + \beta_1 * age + \beta_2 * IQ + u 
$$

In [4]:
import numpy as np # linear algebra 
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
# stats models: regression fitting via formulas
import statsmodels.formula.api as smf
# stats models: regression fitting via matrices of regression design
import statsmodels.api as sm

In [7]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/wage2.csv')
df

wage  hours   IQ  KWW  educ  exper  tenure  age  married  black  south  \
0     769     40   93   35    12     11       2   31        1      0      0   
1     808     50  119   41    18     11      16   37        1      0      0   
2     825     40  108   46    14     11       9   33        1      0      0   
3     650     40   96   32    12     13       7   32        1      0      0   
4     562     40   74   27    11     14       5   34        1      0      0   
..    ...    ...  ...  ...   ...    ...     ...  ...      ...    ...    ...   
930   520     40   79   28    16      6       1   30        1      1      1   
931  1202     40  102   32    13     10       3   31        1      0      1   
932   538     45   77   22    12     12      10   28        1      1      1   
933   873     44  109   25    12     12      12   28        1      0      1   
934  1000     40  107   32    12     17      18   35        1      0      1   

     urban  sibs  brthord  meduc  feduc     lwage  
0        1     1      2.0    8.0    8.0  6.645091  
1        1     1      NaN   14.0   14.0  6.694562  
2        1     1      2.0   14.0   14.0  6.715384  
3        1     4      3.0   12.0   12.0  6.476973  
4        1    10      6.0    6.0   11.0  6.331502  
..     ...   ...      ...    ...    ...       ...  
930      0     0      1.0   11.0    NaN  6.253829  
931      1     7      7.0    8.0    6.0  7.091742  
932      0     9      NaN    7.0    NaN  6.287858  
933      0     1      1.0    NaN   11.0  6.771935  
934      0     5      1.0    NaN    NaN  6.907755  

[935 rows x 17 columns]

In [8]:
# специфицируем модель через формулу
wage_eq1 = smf.ols(formula='np.log(wage)~age+IQ', data=df).fit()

wage_eq1.params.round(3)

Intercept    5.077
age          0.024
IQ           0.009
dtype: float64

- Увеличение результата теста IQ на 1 балл прирощяет месячную зарплату на 0.9%
- Каждый год месячная зарплата человека в среднем увеличивается на 2.4%

## модель 2 

$$ 
log(wage) = \beta_0 + \beta_1 * age + \beta_2 * IQ + ...  + u 
$$

In [9]:
df[['lwage', 'age', 'IQ', 'south', 'urban', 'married', 'KWW']].describe()

lwage         age          IQ       south       urban     married  \
count  935.000000  935.000000  935.000000  935.000000  935.000000  935.000000   
mean     6.779004   33.080214  101.282353    0.341176    0.717647    0.893048   
std      0.421144    3.107803   15.052636    0.474358    0.450385    0.309217   
min      4.744932   28.000000   50.000000    0.000000    0.000000    0.000000   
25%      6.505783   30.000000   92.000000    0.000000    0.000000    1.000000   
50%      6.807935   33.000000  102.000000    0.000000    1.000000    1.000000   
75%      7.056175   36.000000  112.000000    1.000000    1.000000    1.000000   
max      8.032035   38.000000  145.000000    1.000000    1.000000    1.000000   

              KWW  
count  935.000000  
mean    35.744385  
std      7.638788  
min     12.000000  
25%     31.000000  
50%     37.000000  
75%     41.000000  
max     56.000000

In [10]:
# специфицируем модели через формулу
wage_eq1 = smf.ols(formula='lwage~age+IQ+south+urban+married+KWW', data=df).fit()
wage_eq1.params.round(3)

Intercept    5.126
age          0.014
IQ           0.007
south       -0.101
urban        0.165
married      0.191
KWW          0.007
dtype: float64

In [12]:
wage_eq1.nobs

935.0

- Увеличение результата теста IQ на 1 балл прирощяет месячную зарплату на 0.7%
- Каждый год месячная зарплата человека в среднем увеличивается на 1.4%